lib/index.js

```diff
@@ -204,34 +208,66 @@ Amplitude.prototype.track = function(track) {
     window.amplitude.logEvent(event, props);
   }
 
-  // also track revenue
-  var revenue = track.revenue();
-  if (revenue) {
-    // use logRevenueV2 with revenue props
-    if (this.options.useLogRevenueV2) {
-      var price = props.price;
-      var quantity = props.quantity;
-      if (!price) {
-        price = revenue;
-        quantity = 1;
-      }
+  // Ideally, user's will track revenue using an Order Completed event.
+  // However, we have previously setRevenue for any event given it had a revenue property.
+  // We need to keep this behavior around for backwards compatibility.
+  if (track.revenue()) this.setRevenue(mapRevenueAttributes(track));
```

Comment:
this line to be modified once you add the 2nd param of some `opts` or `config` that lets us know that we should not call revenue so that we can still send `revenue` and `total` as custom props to maintain backward compat